extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/ByteBufferAsyncResponseTransformer.java

```diff
@@ -0,0 +1,89 @@
+package io.deephaven.parquet.table.util;
+
+import io.deephaven.base.verify.Assert;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.core.async.AsyncResponseTransformer;
+import software.amazon.awssdk.core.async.SdkPublisher;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * An {@link AsyncResponseTransformer} that transforms a response into a {@link ByteBuffer}.
+ * This class is inspired from {@link software.amazon.awssdk.core.internal.async.ByteArrayAsyncResponseTransformer}
+ * but avoids a number of extra copies done by the former.
+ *
+ * @param <ResponseT> POJO response type.
+ */
+public final class ByteBufferAsyncResponseTransformer<ResponseT> implements AsyncResponseTransformer<ResponseT, ByteBuffer> {
+
+    private volatile CompletableFuture<ByteBuffer> cf;
+    private final ByteBuffer byteBuffer;
+
+    ByteBufferAsyncResponseTransformer(final int bufferSize) {
+        // TODO Can be improved with a buffer pool
```

Comment:
Need to work on this.

Reply:
We talked about pool implementation options, and you now have sample code.